### PR TITLE
[luci-interpreter] Support ExpandDims op in luci-interpreter

### PR DESCRIPTION
**Pull request by BalyshevArtem**
This commit adds support for ExpandDims op in luci-interpreter.

issue: #8781

ONE-DCO-1.0-Signed-off-by: Artem Balyshev a.balyshev@partner.samsung.com

### CODE REVIEW

File: compiler/luci-interpreter/src/kernels/ExpandDims.cpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+namespace luci_interpreter
+{
+namespace kernels
+{
+
+namespace
+{
+
+} // namespace
+
```
Comment (seanshpark) [inline compiler/luci-interpreter/src/kernels/ExpandDims.cpp:29]: maybe unnecessary code?

File: compiler/luci-interpreter/src/loader/nodes/ExpandDims.cpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+namespace luci_interpreter
+{
+
+std::unique_ptr<Kernel> build_kernel_CircleExpandDims(const luci::CircleNode *circle_node,
+                                                      KernelBuilderHelper &helper)
+{
+  const auto *node = dynamic_cast<const luci::CircleExpandDims *>(circle_node);
+  if (node == nullptr)
+    throw std::runtime_error("wrong builder for operation");
+
```
Comment (seanshpark) [inline compiler/luci-interpreter/src/loader/nodes/ExpandDims.cpp:30]: maybe use `loco::must_cast` ?

File: compiler/luci-interpreter/src/kernels/ExpandDims.cpp [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+ExpandDims::ExpandDims(const Tensor *input, const Tensor *axis, Tensor *output)
+  : Kernel({input, axis}, {output})
+{
+}
+
+void ExpandDims::configure()
+{
+  LUCI_INTERPRETER_CHECK(axis()->element_type() == loco::DataType::S32 or
+                         axis()->element_type() == loco::DataType::S64);
```
Comment (jinevening) [inline compiler/luci-interpreter/src/kernels/ExpandDims.cpp:38]: This check is duplicate with the exception in the below switch statement.

File: compiler/luci-interpreter/src/kernels/ExpandDims.cpp (hunk 1 above)
Comment (BalyshevArtem) [reply compiler/luci-interpreter/src/kernels/ExpandDims.cpp]: Remove it, thanks

File: compiler/luci-interpreter/src/loader/nodes/ExpandDims.cpp (hunk 2 above)
Comment (BalyshevArtem) [reply compiler/luci-interpreter/src/loader/nodes/ExpandDims.cpp]: Done

File: compiler/luci-interpreter/src/kernels/ExpandDims.cpp (hunk 3 above)
Comment (BalyshevArtem) [reply compiler/luci-interpreter/src/kernels/ExpandDims.cpp]: Remove it, thanks

File: compiler/luci-interpreter/src/kernels/ExpandDims.cpp [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+      break;
+    default:
+      throw std::runtime_error("Unsupported type.");
+  }
+
+  const auto input_shape = input()->shape();
+
+  if (axis_value < 0)
+  {
+    axis_value += input_shape.num_dims() + 1;
```
Comment (seanshpark) [inline compiler/luci-interpreter/src/kernels/ExpandDims.cpp:50]: is adding `+1` correct? all other Ops just add `input_shape.num_dims()`

Comment (seanshpark) [reply compiler/luci-interpreter/src/kernels/ExpandDims.cpp:50]: or is it for output shape is +1 in dims ?

**Review by jinevening — APPROVED** [review]
LGTM Thank you

File: compiler/luci-interpreter/src/kernels/ExpandDims.cpp (hunk 4 above)
Comment (BalyshevArtem) [reply compiler/luci-interpreter/src/kernels/ExpandDims.cpp:50]: > is adding `+1` correct? all other Ops just add `input_shape.num_dims()`

Yes, this is correct behavior due to for output shape is +1 in dims. You can find the same in https://github.com/tensorflow/tensorflow/blob/200038790349b841789783530a4cb8c58d169ef9/tensorflow/lite/kernels/expand_dims.cc#L37

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Pull request merged**